main/}bedrock.cube.view.create.pro

```diff
@@ -1,645 +1,649 @@
+        nIsDelimiter = 0;
+
+        ## Check dimension delimiter first
+        # If the delimiter is more than 1 character peek ahead the same amount
+        # Ignore the first character
+        sDelim = sChar;
+        nCount = LONG(sDelimDim) - 1;
+        If( nCount > 0 & nChar + nCount <= nCharCount );
+          # Add the extra characters
+          sDelim = sDelim | SUBST( sFilter, nChar + 1, nCount);
+          # Move to the end of the delimter
+          nAddExtra = nCount;
+        EndIf;
+
+        If( sDelim @= sDelimDim );
+          nIsDelimiter = 1;
+          sChar = sDelim;
+        Else;
+          # Reset extra chars
+          nAddExtra = 0;
+
+          ## Check element delimiter
+
+          # If the delimiter is more than 1 character peek ahead the same amount
+          # Ignore the first character
+          sDelim = sChar;
+          nCount = LONG(sDelimElem) - 1;
+          If( nCount > 0 & nChar + nCount <= nCharCount );
+            # Add the extra characters
+            sDelim = sDelim | SUBST( sFilter, nChar + 1, nCount);
+            # Move to the end of the delimter
+            nAddExtra = nCount;
+          EndIf;
+
+          If( sDelim @= sDelimElem );
+            nIsDelimiter = 1;
+            sChar = sDelim;
+          Else;
+            # Reset extra chars
+            nAddExtra = 0;
+          EndIf;
+
+        EndIf;
+
+        If ( nIsDelimiter = 1 );
+
+          If( sLastDelim @= '' % sLastDelim @= sDelimDim );
+            sMessage = 'An element delimiter must follow a dimension name: ' |  sChar | ' (' | NumberToString(nChar) | ')';
+            nErrors = nErrors + 1;
+            LogOutput( cMsgErrorLevel, Expand( cMsgErrorContent ) );
+            #ProcessError();
+          EndIf;
+
+          sElement = sWord;
+
+          If( DIMIX( sDimension, sElement ) = 0 );
+              # The element does not exist in the dimension. Cancel process
+              sMessage = 'Element: ' | sElement | ' in dimension ' | sDimension | ' does not exist';
+              nErrors = nErrors + 1;
+              LogOutput( cMsgErrorLevel, Expand( cMsgErrorContent ) );
+              #ProcessError();
+          EndIf;
+          
+          sElement = DimensionElementPrincipalName(sDimension,sElement);
+
+          If ( (pSuppressConsol = 1 % pIncludeDescendants=1) & DTYPE( sDimension, sElement) @= 'C'  );
+              # Add all N level elements to the subset
+              # Loop through all elements and check if it is an ancestor
+              sMessage = 'Element ' | sElement | ' is consolidated' ;
+              IF ( pLogoutput = 1 );
+                LogOutput( cMsgInfoLevel, Expand( cMsgInfoContent ) );
+              EndIf;
+              nElCount = DIMSIZ ( sDimension );
+              n = 1;
+              WHILE ( n <= nElCount );
+                  sEl = DIMNM( sDimension, n );
+                  If( ElIsAnc(sDimension, sElement, sEL) = 1 );
+                      If( pSuppressConsolStrings = 0 );

```

Comment:
I believe this needs to be changed to `If( pSuppressConsolStrings = 0  % pIncludeDescendants = 1 );`

Reply:
I must be missing something, been a long few days, apologies.

The way I understand it is that the check on line 464 would include all the intermediate consolidations if `pSuppressConsolStrings = 0`, which is kinda what I'd expect.  I read `pIncludeDescendants = 1 and pSuppressConsolStrings =0` as "add all descendants of selected totals and not suppress strings on all elements". 
For the same example cube, running it with an intermediate consolidation added in the dimension and  view create code not changed copies the intermediate values as well:
<img width="498" alt="image" src="https://github.com/cubewise-code/bedrock/assets/30726099/29cf85a5-39a6-450e-a750-c67752e6f4be">
I'm running the following copy syntax:
``` 
ExecuteProcess( '}bedrock.cube.data.copy'
        , 'pLogOutput', 1
        ,'pStrictErrorHandling', 1
        ,'pCube', 'test'
        , 'pSrcView', ''
        , 'pTgtView', ''
        , 'pFilter', 'dim1¦Total'
        ,'pFilterParallel', ''
        , 'pParallelThreads', 0
        ,'pEleMapping', 'dim2¦ version1-> version2'
        , 'pMappingDelim', '->'
        ,'pDimDelim', '&'
        , 'pEleStartDelim', '¦'
        , 'pEleDelim', '+'
        ,'pFactor', 1
        , 'pSuppressConsol', 0
        ,'pSuppressConsolStrings', 0
        , 'pSuppressRules', 0
        , 'pSuppressZero', 1
        ,'pIncludeDescendants',1
        , 'pCumulate', 0
        ,'pZeroTarget', 1
        ,'pZeroSource', 0
        ,'pTemp', 1
        , 'pCubeLogging', 0
        , 'pSandbox', ''
        , 'pFile', 1
        , 'pSubN', 0
    );
```

Reply:
Yes you're right. Don't know what I was thinking.